test/meetingreadinesschecker/DefaultMeetingReadinessChecker.test.ts

```diff
@@ -151,6 +153,19 @@ describe('DefaultMeetingReadinessChecker', () => {
     getRTCPeerConnectionStats(_selector?: MediaStreamTrack): Promise<RTCStatsReport> {
       return !this.noRTCConnection ? new RTCPeerConnection().getStats() : null;
     }
+
+    update(_options: { needsRenegotiation: boolean }): boolean {
+      this.sendClientMetricReport();
+      return false;
+    }
+
+    async sendClientMetricReport(): Promise<void> {
+      const clientMetricReport = new ClientMetricReport(logger);
+      clientMetricReport.rtcStatsReport = await new RTCPeerConnection().getStats();
+      this.forEachObserver(observer => {
+        Maybe.of(observer.metricsDidReceive).map(f => f.bind(observer)(clientMetricReport));
+      });
+    }
```

Comment:
For understanding, why do we need these to methods? I didn't find the equivalent ones in the previous version.

Reply:
A new test will send the fake report during the start and update actions. The update action happens when someone joins/leaves a meeting or toggles a video.

Reply:
Like Kyu said, in the part we directly call `getStats` via the deprecated API so its easier to mock. Now we migrate to use the `metricsDidReceive` event I need some way to trigger it. So I create a new method in this test audio video controller to trigger it.

Reply:
Got it! Thanks for your explanation.